lib/bcdice/game_system/Illusio.rb

```diff
@@ -33,56 +33,47 @@ def initialize(command)
       end
 
       register_prefix(
-        '(\d+)?IL([1-6])?([1-6])?([1-6])?([1-6])?([1-6])?([1-6])?(P)?'
+        '(\d+)?IL([1-6]{0,6})(P)?'
       )
 
       def eval_game_system_specific_command(command)
-        if command =~ /(\d+)?IL([1-6])?([1-6])?([1-6])?([1-6])?([1-6])?([1-6])?(P)?$/i
-          diceCount = (Regexp.last_match(1) || 1).to_i
-          blockNo = [(Regexp.last_match(2) || 0).to_i, (Regexp.last_match(3) || 0).to_i, (Regexp.last_match(4) || 0).to_i, (Regexp.last_match(5) || 0).to_i, (Regexp.last_match(6) || 0).to_i, (Regexp.last_match(7) || 0).to_i]
-          blockNo.delete(0)
-          blockNo = blockNo.sort
-          blockNo = blockNo.uniq
-          isParry = !Regexp.last_match(8).nil?
-
-          return checkRoll(diceCount, blockNo, isParry)
-        end
+        m = command.match(/(\d+)?IL([1-6]{0,6})(P)?$/i)
+        return nil unless m
+
+        dice_count = (m[1] || 1).to_i
+        block_no = (m[2] || "").split(//).map(&:to_i).uniq.sort
```

Comment:
`split(//)` を `each_char` にすると、意図が分かりやすいです。

```suggestion
        block_no = (m[2] || "").each_char.map(&:to_i).uniq.sort
```